lib/fpm/package/freebsd.rb

```diff
@@ -0,0 +1,117 @@
+require "backports" # gem backports
+require "fpm/package"
+require "fpm/util"
+require "archive/tar/minitar"
+require "digest"
+require "fileutils"
+require "xz"
+
+class FPM::Package::FreeBSD < FPM::Package
+  def self.default_abi
+    abi_name = %x{uname -s}.chomp
+    abi_version = %x{uname -r}.chomp.split(".")[0]
+    abi_arch = %x{uname -m}.chomp
+
+    [abi_name, abi_version, abi_arch].join(":")
+  end
+
+  option "--abi", "ABI",
+         "Sets the FreeBSD abi pkg field to specify binary compatibility.",
+         :default => default_abi
+
+  option "--origin", "ABI",
+         "Sets the FreeBSD 'origin' pkg field",
+         :default => "fpm/<name>"
+
+  def output(output_path)
+    output_check(output_path)
+
+    # Build the packaging metadata files.
+    files = {}
+    self.files.each do |f|
+      files[f] = Digest::SHA1.file(File.join(staging_path, f)).hexdigest
+    end
+
+    pkg_origin = attributes[:freebsd_origin]
+    if pkg_origin == "fpm/<name>"  # fill in default
+      pkg_origin = "fpm/#{name}"
+    end
+
+    pkgdata = {
+      "abi" => attributes[:freebsd_abi],
+      "name" => name,
+      "version" => version,
+      "comment" => description,
+      "desc" => description,
+      "origin" => pkg_origin,
+      "maintainer" => maintainer,
+      "www" => url,
+      # prefix is required, but it doesn't seem to matter
+      "prefix" => "/",
+    }
+
+    # Write +COMPACT_MANIFEST, without the "files" section.
+    File.open(File.join(staging_path, "+COMPACT_MANIFEST"), "w+") do |file|
```

Comment:
recommend using the `staging_path(xyz)` shorthand to avoid File.join (for readability)